src/main/scala/com/exasol/spark/DefaultSource.scala

```diff
@@ -140,16 +143,20 @@ class DefaultSource
     newDF.rdd.foreachPartition(iter => writer.insertPartition(iter))
   }
 
-  private[this] def createDFTable(
+  // Creates an Exasol table that match Spark dataframe
+  private[this] def createExasolTable(
     df: DataFrame,
     tableName: String,
     manager: ExasolConnectionManager
-  ): Unit = {
-    if (!manager.config.create_table) {
+  ): Unit =
+    if (manager.config.create_table || manager.config.drop_table) {
+      manager.createTable(tableName, Types.createTableSchema(df.schema))
+    } else {
       throw new UnsupportedOperationException(
         s"""
            |Table $tableName does not exist and cannot be created. Please enable table
```

Comment:
Could split this up into two different exceptions? One if the table exists and another one if it could not be created for other reasons. What confuses me is `Table $tableName does not exist ` ... `Or drop existing table`. Since if it does not exist, I can't drop it...

Reply:
Great suggestion!